src/Sylius/Bundle/CoreBundle/EventListener/Workflow/OrderCheckout/ControlPaymentStateListener.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\EventListener\Workflow\OrderCheckout;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\StateResolver\StateResolverInterface;
+use Symfony\Component\Workflow\Event\CompletedEvent;
+use Webmozart\Assert\Assert;
+
+final class ControlPaymentStateListener
```

Comment:
```suggestion
final class ControlOrderPaymentStateListener
```
or even
```suggestion
final class ResolveOrderPaymentStateListener
```

Reply:
👍 For the second suggestion. ControlSmthListener sounds more like a guard to me.